orderapi/src/main/java/fi/hel/verkkokauppa/order/service/order/OrderService.java

```diff
@@ -131,6 +131,20 @@ public Order createByParams(String namespace, String user) {
         return order;
     }
 
+    public Order createByParams(String namespace, String user, LocalDateTime lastValidPurchaseDateTime) {
+        LocalDateTime createdAt = DateTimeUtil.getFormattedDateTime();
+        String orderId = generateOrderId(namespace, user, createdAt);
+        Long incrementId = this.incrementId.generateOrderIncrementId();
+        Order order = new Order(orderId, namespace, user, createdAt, incrementId);
+        if (StringUtils.isNotEmpty(lastValidPurchaseDateTime.toString())) {
```

Comment:
Can this cause NPO if lastValidPurchaseDateTime is null?

Reply:
I do not think we need to even check the .toString() portion here. If we have an instance of lastValidPurchaseDateTime, then the toString() is not empty. So only **lastValidPurchaseDateTime != null** should be enough in this comparison.

Reply:
Thanks both for the remarks. What Martin added seems to hold true, so I'll change this to be just a null check.